src/assets/scripts/client/airport/MapCollection.js

```diff
@@ -0,0 +1,173 @@
+     *
+     * @for MapCollection
+     * @property hasMaps
+     * @return {boolean}
+     */
+    get hasMaps() {
+        return this.length !== 0;
+    }
+
+    /**
+     * A flag indicating whether the `MapCollection` has any visible maps.
+     *
+     * @for MapCollection
+     * @property hasMaps
+     * @return {boolean}
+     */
+    get hasVisibleMaps() {
+        // for means we can return on finding the first match, rather
+        // than walking the whole array
+        for (let i = 0; i < this.length; i++) {
+            const map = this._items[i];
+
+            if (!map.isHidden && map.hasLines) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Public fascade for `#_items`
+     *
+     * @for MapCollection
+     * @property maps
+     * @return {array<MapModel>}
+     */
+    get maps() {
+        return this._items;
+    }
+
+    /**
+     * A list of all the map lines for the visible maps in the `MapCollection`
+     *
+     * @for MapCollection
+     * @method getVisibleMapLines
+     * @return {array<object>}
+     */
+    getVisibleMapLines() {
+        return this._items.reduce((sum, map) => {
+            if (map.isHidden || !map.hasLines) {
+                return sum;
+            }
+        
+            return [
+                ...map.lines,
+                ...sum
+            ];
+        }, []);
+    }
+
+    /**
+     * A list of all the `MapModel` names in the `MapCollection`
+     *
+     * @for MapCollection
+     * @method getMapNames
+     * @returns {array<string>}
+     */
+    getMapNames() {
+        return this._items.map((map) => {
+            return map.name;
+        });
+    }
+
+    /**
+     * Initialize the instance
+     *
+     * @for MapCollection
+     * @method _init
+     * @param mapJson {object}
+     * @param airportPositionModel {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    _init(mapJson, airportPositionModel, magneticNorth) {
+        this._buildMapModels(mapJson, airportPositionModel, magneticNorth);
+    }
+
+    /**
+     * Tear down the instance and destroy any instance property values
+     *
+     * @for MapCollection
+     * @method destroy
+     */
+    destroy() {
+    }
+
+    /**
+     * @for MapCollection
+     * @method _buildMapModels
+     * @param mapJson {object}
+     * @param airportPositionModel {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    _buildMapModels(mapJson, airportPositionModel, magneticNorth) {
+        // Need to use forEach, as the value being passed can be either
+        // an array of map objects, or a dictionary of key -> object[]
+        _forEach(mapJson, (map, key) => {
+            if (!_has(map, 'lines')) {
+                // Handle the existing map dictionary format too
+                map = {
+                    name: `Legacy-${key}`,
+                    lines: map
```

Comment:
Clever! Typically, I strongly prefer to change all of the (currently active in the load list) airports to conform to the new standard, and consciously do _not_ support the old ones, for the sake of keeping them as consistent as possible. We used to have some crazy stuff going on in three or four airports, and moving to a mindset of consistency has made it a lot easier for everyone to share the work of updating the airports over time.

Just a thought, this is great for now. Maybe I can update the airports and remove this in a few days.

Reply:
I agree, this was more a stopgap so I didn't have to change multiple airport files before a format was agreed upon. Especially in light of your other comment.

I'm happy to go through and update all the files, it's not too onerous and I'd assumed it was part of the branch.